pkg/operator/controller/gatewayclass/subscription.go

```diff
@@ -115,3 +122,40 @@ func subscriptionChanged(current, expected *operatorsv1alpha1.Subscription) (boo
 
 	return true, updated
 }
+
+// ensureServiceMeshOperatorInstallPlan attempts to ensure that the install plan for the appropriate OSSM operator
+// version is approved.
+func (r *reconciler) ensureServiceMeshOperatorInstallPlan(ctx context.Context) (bool, *operatorsv1alpha1.InstallPlan, error) {
+	currentInstallPlan, err := r.getCurrentInstallPlan(ctx)
+	if err != nil {
+		return false, nil, err
+	} else if currentInstallPlan == nil {
+		return false, nil, nil
+	}
+	if !currentInstallPlan.Spec.Approved {
+		currentInstallPlan.Spec.Approved = true
+		if err := r.client.Update(ctx, currentInstallPlan); err != nil {
+			return false, nil, fmt.Errorf("Failed to update %s/%s: %w", currentInstallPlan.Namespace, currentInstallPlan.Name, err)
+		}
+		return true, currentInstallPlan, nil
+	}
+	return false, currentInstallPlan, nil
+}
+
+func (r *reconciler) getCurrentInstallPlan(ctx context.Context) (*operatorsv1alpha1.InstallPlan, error) {
```

Comment:
Might be better to use the function name we normally use.
```suggestion
func (r *reconciler) currentInstallPlan(ctx context.Context) (*operatorsv1alpha1.InstallPlan, error) {
```

Reply:
fixed